ldb_rotating_reader.go

```diff
@@ -0,0 +1,140 @@
+package ctlstore
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/segmentio/ctlstore/pkg/ldb"
+	"github.com/segmentio/stats/v4"
+	"sync/atomic"
+	"time"
+)
+
+// LDBRotatingReader reads data from multiple LDBs on a rotating schedule.
+// The main benefit is relieving read pressure on a particular ldb file when it becomes inactive,
+// allowing sqlite maintenance
+type LDBRotatingReader struct {
+	active         int32
+	dbs            []RowRetriever
+	schedule       []int8
+	now            func() time.Time
+	tickerInterval time.Duration
+}
+
+type RotationFrequency int
+
+const (
+	// Every30 rotate on 30 minute mark in an hour
+	Every30 RotationFrequency = 30
+	// Every20 rotate on 20 minute marks in an hour
+	Every20 RotationFrequency = 20
```

Comment:
RotationFrequency 20 will result in odd number of shifts, meaning ldb-0 will always have twice reading load of ldb-1, am I understand it correctly?